docs/development-guide/project-structure/components/user.md

```diff
@@ -0,0 +1,51 @@
+# User
+
+The User class is part of Gaffer's [access module](https://github.com/gchq/Gaffer/tree/master/core/access).
+When interacting with Gaffer you use a [User](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/user/User.html).
+
+## User Object
+
+The `User` object contains:
+
+- `userId` - Unique identifier for the user
+- `dataAuths` - User authorisations for accessing data
+- `opAuths` - User authorisations for running different Gaffer operations
+
+The `execute` methods of [`Graph`](graph.md) require you to pass in an instance of `User`.
+The examples in this documentation usually use the default user created with `new User()`.
+
+For users of Gaffer, interacting in Java is insecure as you are able to set any authorisations you want to. However, as
+you are writing Java it is assumed that you have access to the `store.properties` file providing the connection details
+(including password) for the cluster where your data is stored (e.g. Accumulo). Therefore, you could bypass Gaffer and
+read all the data directly from disk if you wanted to.
+
+The security layer for Gaffer is currently only enforced by the REST API. We recommend restricting users such that they
+do not have access to the cluster and by only allowing users to connect to Gaffer via the REST API. In the REST API the
+`User` object is constructed via a [`UserFactory`](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/rest/factory/UserFactory.html).
+Currently, we only provide one implementation of this, the [`UnknownUserFactory`](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/rest/factory/UnknownUserFactory.html).
```

Comment:
Where exactly do you set the `UserFactory` that is in use? I assume its one of the config files?

Reply:
Great question, I'll try and document this. I think it is somewhere in the rest config yes.

Reply:
I've added a line covering this. However this aspect should be covered in more detail elsewhere in the docs. There are quite a few other REST related properties in [common-rest `SystemProperty.java`](https://github.com/gchq/Gaffer/blob/298604ff4696c6f7fb070125f719cb27ebcb0db4/rest-api/common-rest/src/main/java/uk/gov/gchq/gaffer/rest/SystemProperty.java#L34-L44) which are not covered anywhere at all. Perhaps an issue could be created to create a new page to cover these, which could then be linked from this page.

Reply:
[New issue created](https://github.com/gchq/gaffer-doc/issues/475) to add more detail on REST properties in general.